src/Chapter2.hs

```diff
@@ -728,7 +736,11 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate []     = []
+smartReplicate (x:xs) = replicate x x ++ smartReplicate xs
+
+smartReplicate2 :: [Int] -> [Int]
+smartReplicate2 xs = concat $ map (\x -> replicate x x) xs
```

Comment:
Oh, sorry, I haven't noticed this one 😄 
Yes, this is correct as well, you can also use `concatMap` 👍🏼